### PR TITLE
Add UDF/SPROCs registration in local testing framework

### PR DESCRIPTION
**Pull request by sfc-gh-melnacouzi**
- Add UDF/SPROCs registration in local testing framework
- Switch to using  Session.builder.getOrCreate() so it works in multiple environments without changes.

### CODE REVIEW

File: streamlit-python/local_test_env.yml [hunk 1]
```diff
@@ -4,9 +4,10 @@ channels:
   - snowflake
 dependencies:
   - python=3.8
-  - snowflake-snowpark-python
+  - snowflake-snowpark-python>=1.15.0
```
Comment (sfc-gh-melnacouzi) [inline streamlit-python/local_test_env.yml:7]: pending snowpark release

**Comment by sfc-gh-bdufour** [comment]
LGTM, but let's do our due diligence again once the Snowpark bug is fixed. Will keep this on the back burner until then.

File: streamlit-python/local_test_env.yml [hunk 2]
```diff
@@ -4,9 +4,10 @@ channels:
...
   - pip
   - pip:
       - pytest
+      - snowflake-cli-labs>=2.0.0
```
Comment (sfc-gh-cgorrie) [inline streamlit-python/local_test_env.yml:11]: Why is this required?

Comment (sfc-gh-melnacouzi) [reply streamlit-python/local_test_env.yml:11]: if you are in testing mode, you might want to use Snow CLI to deploy your app. It's annoying to have to deactivate and reactivate your conda environment to use CLI, or to test locally.

Comment (sfc-gh-melnacouzi) [reply streamlit-python/local_test_env.yml:11]: I can remove the version requirement though to avoid confusion

File: streamlit-python/src/module-ui/test/test_ui.py [hunk 3]
```diff
@@ -1,46 +1,28 @@
...
-@patch('snowflake.snowpark.functions.call_udf')
-@patch('snowflake.snowpark.context.get_active_session')
-def test_streamlit(get_active_session, call_udf, call_sproc, session: Session):
-    get_active_session.return_value = session
-    call_udf.side_effect = mock_call_udf
-    call_sproc.side_effect = mock_call_sproc
+@fixture(autouse=True)
+def set_up(session: Session):
+    session.sproc.register(add.increment_by_one_fn, name='core.increment_by_one')
+    session.udf.register(add.add_fn, name='core.add')
```
[the remark this answers is not in the document]
Comment (sfc-gh-bgoel) [reply streamlit-python/src/module-ui/test/test_ui.py:17]: Is this still an issue? If yes, we should wait before merging something with a bug.

[the remark this answers is not in the document]
Comment (sfc-gh-melnacouzi) [reply streamlit-python/src/module-ui/test/test_ui.py:17]: no, this is fixed.

File: streamlit-python/local_test_env.yml (hunk 1 above)
Comment (sfc-gh-melnacouzi) [reply streamlit-python/local_test_env.yml]: released.

**Review by sfc-gh-bgoel — APPROVED** [review]
